### DISCUSSION TITLE
Monitor ui.upload to detect when an uploaded file gets removed

### DISCUSSION THREAD

**Discussion by Rjdrenth** [zauberzeug/nicegui#3133] [Ideas / Feature Requests]
Hi, thanks for the great work on NiceGUI! It is by far my favourite python GUI library :)

I was working on an app that lets users upload a file and I display detected properties / issues with the actual contents of the file in a table. The idea is that the user knows whether the file is valid or not before proceeding to the next. 

What I need is to detect when the user deletes an uploaded file so I can properly propagate those changes in the rest of my application. The [QUploader](https://quasar.dev/vue-components/uploader/) apparently has the following event: `@removed: (files) => void`.
Unfortunately it seems that NiceGUI does not capture this event, or at least it's not clear to me how.

Is this possible? If not, could this be added?

Alternatively, if someone points me in the right direction of where those changes can be made I'd be willing to give it a go to make a PR. My Javascript sucks though, so I'd need to be pointed in the direction of a a relatively comparable piece of code for inspiration.

**Comment by Rjdrenth** [comment]
Let me answer my own question, yes, this is very possible. 

`ui.upload(..).on("removed", lambda event: print(event)` allows me to track this. The only downside is that it does not give me the exact filename, but rather a file key which prepends and appends something to the filename, which is not ideal, but manageable.

**Reply by jmart5** [reply]
This works great. Thanks!
